### ISSUE TITLE
Error Reports are not very helpful

### ISSUE THREAD

**Issue by philip30** [neulab/xnmt#495]
Let's maybe not printing the graph on the console as it can easily fill up your console and hard to scroll it up. Maybe we can dump it in a log, but not on the screen?

Also, let's maybe print the source and target using the real words not the ID. Using ID's are not very helpful.

**Comment by philip30** [comment]
How about making error logger?

```python
err = logging.getLogging("error")

# In other files
from xnmt import err

.....
err.info(print_graph())
```

**Comment by msperber** [comment]
We already have a file_logger that prints only to the log file, so you could try something like this:
```
def log_graph(): file_logger.error(dy.print_text_graphviz()); return "DyNet graph logged to file"
with utils.ReportOnException({"src": src, "trg": trg, "graph": log_graph}):
```

For printing actual words, I think overwriting ```ReadableSentence.__str__()``` would be the best way.

**Comment by neubig** [comment]
I agree that this would be nice to fix. The computation graph is helpful sometimes, but it gets in the way the 95% of the time it's not helpful.

**Comment by msperber** [comment]
fixed by #517.
